triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala

```diff
@@ -290,4 +298,37 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers with Postg
         _ <- assert(resp.status.isSuccess)
       } yield succeed
   }
+
+  it should "fail to start a trigger if a ledger client can't be obtained" in withHttpService(
+    Some(dar)) { (uri: Uri, client: LedgerClient, ledgerProxy: Proxy) =>
+    // Disable the proxy. This means that the service won't be able to
+    // get a ledger connection.
+    ledgerProxy.disable()
+    try {
+      // Request a trigger be started and setup an assertion that
+      // completes with success when the running trigger table becomes
+      // non-empty.
+      val runningTriggersNotEmpty = for {
+        resp <- startTrigger(uri, s"$testPkgId:TestTrigger:trigger", "Alice")
+        aliceTrigger <- parseTriggerId(resp)
+        triggerId <- parseTriggerId(resp)
+      } yield (assertTriggerIds(uri, "Alice", triggerIds => triggerIds.nonEmpty))
+      // Wait a good while (10s) on the assertion to become true. If it
+      // does, indicate the test has failed (because, since the trigger
+      // can't be initialized for the lack of a viable ledger client
+      // connection, it should not make its way to the running triggers
+      // table).
+      Await.ready(awaitable = runningTriggersNotEmpty, atMost = 10.seconds)
+      fail("Timeout expected")
```

Comment:
A more descriptive error might be helpful here, like "Trigger added despite disabled ledger client".